File: SOEN2030 数据结构课程实践/实验/2021-10-21 实验六/Exercise2.java
public class Exercise2 {

    public static void main(String[] args) {
        int k = 17;
        int[] lst1 = {2, 3, 4, 5, 9, 10, 17, 19, 22, 23, 25, 26, 27, 30, 30, 33, 39, 40, 42, 44, 46, 46, 47, 48, 48, 50, 54, 55, 56, 57, 63, 67, 68, 70, 72, 73, 76};
        int[] lst2 = {8, 13, 15, 16, 16, 20, 21, 22, 24, 28, 29, 31, 32, 34, 36, 37, 38, 40, 40, 41, 43, 44, 45, 48, 49, 50, 50, 52, 53, 59, 61, 62, 64, 64, 67, 68, 75, 77, 77};
        int i = 0, j = 0;
        int count = 1;
        int result = lst1[0];
        while (count < k) {
            while (lst1[i] < lst2[j] && count < k) {
                if (lst1[i] > result) {
                    result = lst1[i];
                    count++;
                }
                i++;
            }
            while (lst2[j] <= lst1[i] && count < k) {
                if (lst2[j] > result) {
                    result = lst2[j];
                    count++;
                }
                j++;
            }
        }
        System.out.println(result);
    }

}

File: COMS3040 信息检索综合实践/实验/2023-03-08 实验3 页面信息抽取/webpage-information-extraction/src/exercise2.ts
import fs from 'node:fs/promises';
import path from 'node:path';

import * as cheerio from 'cheerio';

import { extractText } from './utils.js';

/**
 * Process all html files in a folder and save the result to another folder.
 * @param sourceFolder
 * @param outputFolder
 * @param logProgress
 */
const exercise2 = async (
  sourceFolder: string,
  outputFolder: string,
  logProgress = false,
) => {
  // Get all file names in the source folder.
  const fileNames = await fs.readdir(sourceFolder);

  for (const fileName of fileNames) {
    const sourcePath = path.join(sourceFolder, fileName);
    const outputPath = path.join(outputFolder, `${fileName.slice(0, -4)}txt`);
    if (logProgress) {
      console.log(`Processing "${sourcePath}" to "${outputPath}"...`);
    }

    // Load html file.
    const html = await fs.readFile(sourcePath, 'utf-8');
    const $ = cheerio.load(html);

    // Extract the main part of the page.
    const body = $('table[bgcolor="#C0C0C0"] tr td') // Find the main table.
      .contents()
      .not('font[color="#FFFFFF"], a') // Filter out advertisement and links.
      .map((_, el) => extractText($, el)) // Extract text from each node.
      .get()
      .join('')
      .replace(/\n{2,}/g, '\n') // Remove redundant newlines.
      .trim() // Remove leading and trailing whitespace.
      .split('相关词语', 2)[0] // Remove words after "相关词语".
      .split('基本解释', 2)
      .join('\n基本解释'); // Add a line break before "基本解释".

    // Save the result to a text file.
    await fs.writeFile(outputPath, body, 'utf-8');

    if (logProgress) {
      console.log('Done.\n');
    }
  }
};

export default exercise2;
